### PR TITLE
Add step orderable elements controller [SCI-6711]

### PR DESCRIPTION
**Pull request by okriuchykhin**
Jira ticket: [SCI-6711](https://biosistemika.atlassian.net/browse/SCI-6711)

### What was done
Add step orderable elements controller [SCI-6711]

[SCI-6711]: https://biosistemika.atlassian.net/browse/SCI-6711?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by aignatov-bio — CHANGES_REQUESTED** [review]
Will we also migrate here TinyMCE images? Now they belong to step. And I think some updates for the TinyMceAssets controller and MarvinJS controller/service, we have all checks with Step as the object.

**Comment by artoscinote** [comment]
Could we call this just StepElement? I think this Orderable part is a bit redundant here.

**Comment by okriuchykhin** [comment]
> Could we call this just StepElement? I think this Orderable part is a bit redundant here.

But it clearly describes purpose of this model

File: app/controllers/step_orderable_elements_controller.rb [hunk 1]
```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+class StepOrderableElementsController < ApplicationController
+  before_action :load_vars
+  before_action :check_manage_permissions
+
+  def update
+    @step.with_lock do
+      element_to_swap = @step.step_orderable_elements.find_by(id: params[:new_position])
```
Comment (aignatov-bio) [inline app/controllers/step_orderable_elements_controller.rb:9]: I think we do not swap elements. We just move one element to the new position and all elements between old and new positions must be moved up/down depending on direction.

**Review by aignatov-bio — CHANGES_REQUESTED** [review]

**Comment by okriuchykhin** [comment]
Obsolete

**Pull request closed without merging**